### PR TITLE
Add models closing, replace queue to deque and add mask 0xFF to cv2.waitKey

### PR DESCRIPTION
**Pull request by vladislav-serdyuk**

### CODE REVIEW

**Review by xImDoctor — CHANGES_REQUESTED** [review]
## Предыдущие правки
Правки по предыдущим замечаниям внесены корректно, использован `deque`, обёртка `try-finally`, маска `0xFF` для `waitKey`, print'ы для текущего `main`.

## Новая критическая проблема
Строки 253-256:
После инициализации 
```python
face_detector_and_emotion_recognizer = DetectFaceAndRecognizeEmotion(face_detector, emotion_recognizer)
```
face_detector_and_emotion_recognizer никогда не None, условие в `finally` всегда ложно и код не выполняется.
```python
    finally:
        if face_detector_and_emotion_recognizer is None:
            face_detector.close()
            emotion_recognizer.reset()
 ```
 
 ### Решение
 Можно ввести флаг `cleanup_needed` перед инициализацией моделей и с его помощью проверять, созданы ли они внутри. Если созданы - очищать в `finally`
 ```python
  cleanup_needed = face_detector_and_emotion_recognizer is None
    
    if cleanup_needed:
        face_detector = FaceDetector(...)
        emotion_recognizer = EmotionRecognizer(...)
        face_detector_and_emotion_recognizer = DetectFaceAndRecognizeEmotion(
            face_detector, emotion_recognizer
        )
    
    try:
        # здесь основной цикл while
    finally:
        if cleanup_needed:
            face_detector_and_emotion_recognizer.face_detector.close()
            face_detector_and_emotion_recognizer.emotion_recognizer.reset()
            del face_detector_and_emotion_recognizer # объект уничтожится после выхода из генератора, но для приложения Streamlit разрушим явно, чтобы избежать утечек
 ```
 
 ## Маскируем ошибку под Neutral 0.0 (строка 176)
 При пересмотре кода заметил критическую проблему
 ```python
         except Exception as e:
            print(f"Ошибка распознавания: {e}")
            import traceback
            traceback.print_exc()

        return "Neutral", 0.0
```
Лучшей практикой будет явный отлов различных типов исключений.
```python
try:
    emotion, scores = self.recognizer.predict_emotions(face_crop, logits=True)
    # обработка
    
except (torch.cuda.OutOfMemoryError, MemoryError):
    # Критично - пробрасываем выше для обработки
    raise
    
except (ValueError, RuntimeError, AttributeError) as e:
    # Ожидаемые проблемы обработки - логируем и fallback
    print(f"Предупреждение при распознавании: {e}")
    return "Neutral", 0.0
    
# Без общего except - если что-то неожиданное, пусть упадёт явно
```
Почему текущая реализация опасна: получили `torch.cuda.OutOfMemoryError`, приложение выбросило `Neutral 0.0` и продолжило работу, повреждена модель - продолжаем работу и так до бесконечного `Neutral` или пока из-за накопления ошибок приложение не упадёт. Хотим явно обрабатывать опасные исключения.


## Некритичные замечания
### 1. Импорт `time` из `main` лучше вынести наверх для соответствия PEP 8
```python
if __name__ == '__main__':
    from time import time
 ```
### 2. Архитектурное улучшение: избавиться от магических чисел в `FaceDetector`
Следует явно вынести `margin = 20` в конструктор `__init__` и на будущее не хранить магических чисел внутри классов. Для бенчмарка подобное было бы нормально, но для гибкости настройки моделей в нашем приложении следует указывать подобные переменные в конструкторе.

**Review by xImDoctor — CHANGES_REQUESTED** [review]
## Комментарий по правкам
Все правки внесены корректно, но осталось переприсваивание магическим числом переменной `margin` (строка 54):
```python
self.margin = 20
```
Нужно убрать переприсваивание значением 20, т.к. `margin` уже задаётся в конструкторе как
```python
self.margin = margin
```
В противном случае при использовании метода `detect`: `margin` всегда будет `20`
